src/main/java/com/avispl/symphony/dal/communicator/other/genericserver/WebClientCommunicator.java

```diff
@@ -102,70 +174,146 @@ public void setURI(String URI) {
 						throw new ResourceNotReachableException(errorMessage);
 					}
 				} else {
-					throw new ResourceNotReachableException(errorMessage);
+					throw new ResourceNotReachableException(errorMessage, exc);
 				}
 			}
 			uriStatusMessage = generateResponseMessage(statusCode);
-			stats.put(URI_STATUS, uriStatusMessage);
+			stats.put(WebClientConstant.URI_STATUS, uriStatusMessage);
 		} else {
-			stats.put(URI_STATUS, NOT_CONFIGURED);
+			stats.put(WebClientConstant.URI_STATUS, WebClientConstant.NOT_CONFIGURED);
 		}
 		extStats.setStatistics(stats);
 		return Collections.singletonList(extStats);
 	}
 
+	/**
+	 * {@inheritDoc}
+	 * <p>
+	 * Get data from uri path
+	 *
+	 * @param uri the uri is the path get from the configuration properties on the symphony portal
+	 * @return String This returns the status code and dataBody if the response get body not null
+	 * @throws Exception if getting information from the Uri failed
+	 */
 	@Override
-	protected void internalInit() throws Exception {
-		super.internalInit();
-		buildBaseUrl();
+	public String doGet(String uri) throws Exception {
+		HttpClient client = this.obtainHttpClient(false);
+
+		String getUri = this.buildRequestUrl(uri);
+		if (this.logger.isTraceEnabled()) {
+			this.logger.debug("Performing a GET operation for " + getUri);
+		}
+
+		StringBuilder stringBuilder = new StringBuilder();
+		HttpResponse response = null;
+		try {
+			response = client.execute(new HttpGet(getUri));
+			// status code
+			stringBuilder.append(response.getStatusLine().getStatusCode());
+
+			HttpEntity httpEntity = response.getEntity();
+			if (isParseContent && httpEntity != null) {
+				// response body
+				String dataBody = EntityUtils.toString(httpEntity);
+				if (!StringUtils.isNullOrEmpty(dataBody)) {
+					stringBuilder.append(statusAndBodySeparator);
+					stringBuilder.append(dataBody);
+				}
+
+				// content type
+				stringBuilder.append(bodyAndContentTypeSeparator);
+				stringBuilder.append(getContentType(httpEntity));
+			}
+		} finally {
+			if (response instanceof CloseableHttpResponse) {
+				((CloseableHttpResponse) response).close();
+			}
+		}
+		return stringBuilder.toString();
 	}
 
 	/**
-	 * WebClientCommunicator doesn't require authentication
+	 * Retrieves {@code {@link #exclude }}
 	 *
-	 * {@inheritDoc}
+	 * @return value of {@link #exclude}
+	 * @since 1.2.0
 	 */
-	@Override
-	protected void authenticate() {
-		// WebClientCommunicator doesn't require authentication
+	public String getExclude() {

```

Comment:
can you please reorder class members according to the following: 

Common convention (slightly enhanced for our specifics) of order of class members – from top to bottom:
-          inner classes/enums
-          static constants (with very rare exceptions, the only static variables should be constants)
-          static methods (public, protected, private)
-          constructor(s)
-          getters/setters (sorted alphabetically)
-          life cycle methods (init/destroy etc)
-          business logic interfaces methods (grouped by interface, then alphabetically in each interface)
-          other methods (protected, private).
 
If class implements more than one business interface, it is recommended to put separators with comments between them.

Reply:
Alphabetical order is not that critical, but general order is something that wouldn't hurt for sure :)